core/lib/src/config/secret_key.rs

```diff
@@ -178,6 +228,70 @@ impl SecretKey {
     {
         ser.serialize_bytes(&[0; 32][..])
     }
+
+    fn cipher(&self, nonce: &[u8]) -> Result<XChaCha20Poly1305, Error> {
+        let (mut prk, hk) = Hkdf::<Sha256>::extract(Some(nonce), self.key.encryption());
+        hk.expand(b"secret_key_data_encryption", &mut prk).map_err(|_| Error::KeyLengthError)?;
+
+        Ok(XChaCha20Poly1305::new(&prk))
+    }
+
+    /// Encrypts the given data.
+    /// Generates a random nonce for each encryption to ensure uniqueness.
+    /// Returns the Vec<u8> of the concatenated nonce and ciphertext.
+    ///
+    /// # Example
+    /// ```rust
+    /// use rocket::config::SecretKey;
+    ///
+    /// let plaintext = "I like turtles".as_bytes();
+    /// let secret_key = SecretKey::generate().unwrap();
+    ///
+    /// let cipher = secret_key.encrypt(&plaintext).unwrap();
+    /// let decrypted = secret_key.decrypt(&cipher).unwrap();
+    ///
+    /// assert_eq!(plaintext, decrypted);
+    /// ```
+    pub fn encrypt<T: AsRef<[u8]>>(&self, value: T) -> Result<Cipher, Error> {
+        let nonce = XChaCha20Poly1305::generate_nonce(&mut OsRng);
+        let cipher = self.cipher(&nonce)?;
+
+        let ciphertext = cipher
+            .encrypt(&nonce, value.as_ref())
+            .map_err(|_| Error::EncryptionError)?;
+
+        // Prepare a vector to hold the nonce and ciphertext
+        let mut encrypted_data = Vec::with_capacity(nonce.len() + ciphertext.len());
+        encrypted_data.extend_from_slice(nonce.as_slice());
+        encrypted_data.extend_from_slice(&ciphertext);
+
+        Ok(Cipher(encrypted_data))
+    }
+
+    /// Decrypts the given encrypted data, encapsulated in a Cipher wrapper.
+    /// Extracts the nonce from the data and uses it for decryption.
+    /// Returns the decrypted Vec<u8>.
+    pub fn decrypt(&self, encrypted: &Cipher) -> Result<Vec<u8>, Error> {
+        let encrypted = encrypted.as_bytes();
+
+        // Check if the length of decoded data is at least the length of the nonce
+        let nonce_len = <XChaCha20Poly1305 as AeadCore>::NonceSize::USIZE;
+        if encrypted.len() <= nonce_len {
+            return Err(Error::EncryptedDataLengthError);
+        }
```

Comment:
This should not need to be checked here. Rather, it should be checked when `Cipher` is constructed.